File: src/main/java/com/controller/AccountController.java
package com.controller;

import com.domain.DbAccount;
import com.service.AccountService;
import com.utils.ResponseResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController // = @Controller + @ResponseBody
//@RequestMapping(value = "/account")
public class AccountController {

    @Autowired
    private AccountService accountService;

    @GetMapping(value = "/login")
    public ResponseResult login(@RequestParam(value = "account") String a,
                           @RequestParam(value = "password") String p){
        DbAccount account = accountService.accountInfo(a);

        if (account == null){
            return ResponseResult.error(1);
        }

        if (!account.getPassword().equals(p)){
            return ResponseResult.error(2);
        }

        return ResponseResult.success(account);
    }

    @PostMapping(value = "/register")
    public ResponseResult register(@RequestParam(value = "account") String a,
                                   @RequestParam(value = "password") String p,
                                   @RequestParam(value = "uid") String uid){

        DbAccount iv = accountService.register(a, p, uid);

        return ResponseResult.success(iv);
    }

    @PostMapping(value = "/changepassword")
    public ResponseResult changePassword(@RequestParam(value = "uid") String uid,
                                         @RequestParam(value = "oldpassword") String op,
                                         @RequestParam(value = "newpassword") String np){

        DbAccount iv = accountService.accountInfoByUid(uid);

        if (iv.getPassword().equals(op)){
            Integer rv = accountService.changePassword(np, iv.getId(), iv);
            if (rv == 0){
                return ResponseResult.error(4, "密码修改失败");
            }

            return ResponseResult.success();
        }
        else{
            return ResponseResult.error(3, "密码错误");
        }
    }

}

File: src/main/java/com/controller/AccountController.kt
package com.controller

import com.domain.entity.DbAccount
import com.service.AccountService
import com.utils.ResponseResult
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

/**
 * class
 *
 * @author Five0314
 * @date create in 下午3:37 2017/12/14
 */
@RestController
@RequestMapping("/account")
class AccountController {
    @Autowired
    private lateinit var accountService: AccountService

    /**
     * 登录
     * @param account 账号
     * @param password 密码
     * @return
     */
    @GetMapping(value = "/login")
    fun login(@RequestParam(value = "account") account: String, @RequestParam(value = "password") password: String) : ResponseResult<DbAccount>{
        return this.accountService.login(account, password)
    }

    /**
     * 注册
     * @param account 账号
     * @param password 密码
     * @return
     */
    @PostMapping(value = "/register")
    fun register(@RequestParam(value = "account") account: String, @RequestParam(value = "password") password: String) : ResponseResult<DbAccount>{

        return this.accountService.register(account, password)
    }

    /**
     * 修改密码
     * @param uid 用户编号
     * @param oldPassword 旧密码
     * @param newPassword 新密码
     * @return
     */
    @PostMapping(value = "/changePassword")
    fun changePassword(@RequestParam(value = "uid") uid: String, @RequestParam(value = "oldPassword") oldPassword: String, @RequestParam(value = "newPassword") newPassword: String) : ResponseResult<DbAccount>{

        return this.accountService.changePassword(uid, oldPassword, newPassword)
    }
}